arch/arm/src/tlsr82/tlsr82_serial.c

```diff
@@ -1558,6 +1558,10 @@ static inline int tlsr82_uart_lowputc(int ch)
 #ifndef CONFIG_TLSR82_SPI_SYSLOG
 int up_putc(int ch)
 {
+  /* To make the syslog be thread safe, add sched_lock()/sched_unlock() */
```

Comment:
This is unrelated to the SERIAL_CONSOLE, even do not enable HAVE_SERIAL_CONSOLE , the problem is still existed, because send byte operation is not atomic (see function `uart_send_byte()`). Maybe a alternative solution is add `enter/leave_critical_section()` in function `uart_send_byte()`, which is similar to the solution show in `arch/arm/src/tms570/tms570_lowputc.c`. What do you think?

Reply:
I posted two separate comments. One related to ifdef wrapper that I think should be applied anyway and the second one related to MT safety. IMO approach with critical section is better because same putc can be used in the ISR as well.

Reply:
Ack, i will do this later today.

Reply:
Done